worldwind/src/commonMain/kotlin/earth/worldwind/layer/RenderableLayer.kt

```diff
@@ -61,22 +68,38 @@ open class RenderableLayer @JvmOverloads constructor(displayName: String? = null
 
     fun addAllRenderables(iterable: Iterable<Renderable>) { for (renderable in iterable) renderables.add(renderable) }
 
-    fun removeRenderable(renderable: Renderable) = renderables.remove(renderable)
+    fun removeRenderable(renderable: Renderable) : Boolean {
+        if (renderables.remove(renderable)) {
+            batchRenderers[renderable::class]?.removeRenderable(renderable)
+            return true
+        }
+        return false
+    }
 
     fun removeRenderable(index: Int): Renderable {
         require(index in renderables.indices) {
             logMessage(ERROR, "RenderableLayer", "removeRenderable", "invalidIndex")
         }
+        val renderable = renderables[index]
```

Comment:
The same here, coud you use the result of removeAt instead of getting it by index.

Reply:
done